sqswatcher/sqswatcher.py

```diff
@@ -122,87 +118,106 @@ def pollQueue(scheduler, q, t, proxy_config):
+                                        'instanceId': instance_id,
+                                        'hostname': hostname
+                                    })
+                                else:
+                                    log.error("Unable to get the hostname for the instance %s." % instance_id)
                             break
                         except ClientError as e:
                             if e.response.get('Error').get('Code') == 'RequestLimitExceeded':
+                                log.debug("Request limit exceeded, waiting other %s seconds..." % wait)
                                 time.sleep(wait)
                                 retry += 1
-                                wait = (wait*2+retry)
+                                wait = wait * 2
                             else:
-                                raise e
+                                log.critical(e.response.get("Error").get("Message"))
+                                break
                         except:
                             log.critical("Unexpected error: %s" % sys.exc_info()[0])
-                            raise
+                            break
+                    message.delete()
 
-                elif (eventType == 'autoscaling:EC2_INSTANCE_TERMINATE') or (eventType == 'EC2 Instance State-change Notification'):
-                    if eventType == 'autoscaling:EC2_INSTANCE_TERMINATE':
-                        instanceId = message_attrs.get('EC2InstanceId')
-                    elif eventType == 'EC2 Instance State-change Notification':
+                elif (event_type == 'autoscaling:EC2_INSTANCE_TERMINATE') or (event_type == 'EC2 Instance State-change Notification'):
+                    if event_type == 'autoscaling:EC2_INSTANCE_TERMINATE':
+                        instance_id = message_attrs.get('EC2InstanceId')
+                    elif event_type == 'EC2 Instance State-change Notification':
                         if message_attrs.get('detail').get('state') == 'terminated':
                             log.info('Terminated instance state from CloudWatch')
-                            instanceId = message_attrs.get('detail').get('instance-id')
+                            instance_id = message_attrs.get('detail').get('instance-id')
                         else:
                             log.info('Not Terminated, ignoring')
                             message.delete()
                             break
 
-                    log.info("instanceId=%s" % instanceId)
-                    try:
-                        item = t.get_item(ConsistentRead=True, Key={"instanceId": instanceId})
-                        if item.get('Item') is not None:
-                            hostname = item.get('Item').get('hostname')
+                    log.info("instance_id=%s" % instance_id)
+                    retry = 0
+                    wait = 15
+                    while retry < 3:
```

Comment:
Shall we export the retry logic and the error handling logic to a common function since it is the same for all event types? This can be a good reason to start breaking this huge function into smaller ones :)

Another nice way to implement retries in python is with annotations as also implemented by this library: https://github.com/rholder/retrying

Reply:
Done